### PR TITLE
improve aio life cycle of quic_api(transport layer)

### PR DESCRIPTION
**Pull request by JaylinYu**

### CODE REVIEW

**Review by lee-emqx — APPROVED** [review]

File: src/supplemental/quic/quic_api.c [hunk 1]
```diff
@@ -1075,12 +1080,18 @@ quic_pipe_close(void *qpipe, uint8_t *code)
 		log_warn("close the QUIC stream!");
 		MsQuic->StreamClose(qstrm->stream);
 	}
+
```
Comment (wanghaEMQ) [inline src/supplemental/quic/quic_api.c:1083]: We need a lock here. Or a data race would happen between pipe_close and pipe_send/pipe_recv on sendq/recvq.

File: src/supplemental/quic/quic_api.c [hunk 2]
```diff
@@ -1075,12 +1080,18 @@ quic_pipe_close(void *qpipe, uint8_t *code)
...
+	// take care of aios
+	while ((aio = nni_list_first(&qstrm->sendq)) != NULL) {
+		nni_list_remove(&qstrm->sendq, aio);
+		nni_aio_abort(aio, NNG_ECANCELED);
```
Comment (wanghaEMQ) [inline src/supplemental/quic/quic_api.c:1087]: The reason code should be NNG_ECLOSED

File: src/mqtt/transport/tcp/mqtt_tcp.c [hunk 3]
```diff
@@ -161,7 +161,7 @@ mqtt_pipe_timer_cb(void *arg)
 	// send PINGREQ with tmaio itself?
 	// nng_msleep(p->keepalive);
 	nni_mtx_lock(&p->mtx);
-	if (!p->busy) {
+	if (!p->busy && !nni_aio_busy(p->qsaio)) {
 		// send pingreq
 		buf[0] = 0xC0;
 		buf[1] = 0x00;
```
Comment (wanghaEMQ) [inline src/mqtt/transport/tcp/mqtt_tcp.c:167]: Did you forget to set the pipe busy?
Should we also check if the aio is busy or not executed?

File: src/mqtt/protocol/mqtt/mqtt_client.c [hunk 4]
```diff
@@ -355,6 +355,15 @@ mqtt_send_msg(nni_aio *aio, mqtt_ctx_t *arg)
 	nni_msg *        msg   = NULL;
 	nni_msg *        tmsg  = NULL;
 
+	if (p == NULL) {
+		// pipe closed, should never gets here
+		nni_println("Sendong msg on a NULL pipe!");
```
[the remark this answers is not in the document]
Comment (JaylinYu) [reply src/mqtt/protocol/mqtt/mqtt_client.c:360]: sure

**Pull request merged**